agent/rpc/middleware/interceptors.go

```diff
@@ -22,27 +23,26 @@ const RPCTypeInternal = "internal"
 const RPCTypeNetRPC = "net/rpc"
 
 var metricRPCRequest = []string{"rpc", "server", "call"}
-var requestLogName = "rpc.server.request"
+var requestLogName = strings.Join(metricRPCRequest, "_")
 
-var NewRPCGauges = []prometheus.GaugeDefinition{
+var OneTwelveRPCSummary = []prometheus.SummaryDefinition{
 	{
 		Name: metricRPCRequest,
-		Help: "Increments when a server makes an RPC service call. The labels on the metric have more information",
+		Help: "Measures the time an RPC service call takes to make in milliseconds. Labels mark which RPC method was called and metadata about the call.",
 	},
 }
 
 type RequestRecorder struct {
 	Logger       hclog.Logger
-	recorderFunc func(key []string, start time.Time, labels []metrics.Label)
+	recorderFunc func(key []string, val float32, labels []metrics.Label)
 }
 
 func NewRequestRecorder(logger hclog.Logger) *RequestRecorder {
-	return &RequestRecorder{Logger: logger, recorderFunc: metrics.MeasureSinceWithLabels}
+	return &RequestRecorder{Logger: logger, recorderFunc: metrics.AddSampleWithLabels}
```

Comment:
`MeasureSinceWithLabels` does cycle work which can take additional time. We have the `elaspsed` time in hand can pass it in as a value instead to avoid any additional error margins.